### PR TITLE
Jbruno/dpt12331 scheduler throttle

### PR DESCRIPTION
**Pull request by JeanetteBruno**
_Problem_
Allow us to throttle the scheduler, limiting the size of jobs it will accept.

_Solution_
Added throttling API to the scheduler:
**Throttle(limit)**: 
* limit == -1->no throttling, 
* limit > -1-> don't accept new job requests if the job's task count puts the scheduler's active and waiting tasks over limit

**GetSchedulerStatus()**:
returns: {receivingJobs: <true/false>, currentTasks: <number of tasks currently running/waiting>, maxTasks: <throttle limit>}

Note: The implementation does not try to resolve race conditions between accepting job requests and setting the throttle limit.

**JIRA: DPT-12331**

### CODE REVIEW

**Comment by dgdonut** [comment]
Before review in detail, pasting comment from ticket:
Suggest we have SetSchedulerStatus to correspond with the Get API, that takes equivalent param (i.e. maxtasks ) for throttling. Also suggest dropping ReceivingJobs since it seems like this could just be interpreted from maxtasks, and the only way we can set this based on the draft is from that value anyway.

**Comment by JeanetteBruno** [comment]
agree with taking out receiving, but not about changing throttle_scheduler to setSchedulerStatus... setSchedulerStatus is too generic, throttle_scheduler is a more intuitive name.

**Comment by JeanetteBruno** [comment]
ignore prior statement about keeping Throttle, changed it to SetSchedulerStatus.

File: binaries/scootapi/main.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ import (
 // 		get_job_status [job id]
 //		watch_job [job id]
 //		run_smoke_test
+//		get_scheduler_status
+//		throttle [limit]
```
Comment (dgdonut) [inline binaries/scootapi/main.go]: change ref to set_scheduler_status

**Review by dgdonut — COMMENTED** [review]
less test failures, lgtm!

File: binaries/scootapi/main.go (hunk 1 above)
Comment (JeanetteBruno) [reply binaries/scootapi/main.go]: done

File: sched/scheduler/stateful_scheduler.go [hunk 2]
```diff
@@ -109,6 +112,7 @@ type SchedulerConfig struct {
 	MaxRequestors           int
 	MaxJobsPerRequestor     int
 	SoftMaxSchedulableTasks int
+	HardMaxSchedulableTasks int
```
Comment (dgdonut) [inline sched/scheduler/stateful_scheduler.go]: Suggest adding to typedef comment and noting it's a suggestion vs a hard cap now (maybe rename the internal var?)

**Review by dgdonut — APPROVED** [review]

**Pull request merged**